first_task.scala

```diff
@@ -0,0 +1,260 @@
+// Databricks notebook source
+import java.sql.Timestamp
+import java.net.{URI, URISyntaxException}
+
+
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.{Dataset, Encoders}
+import org.apache.spark.sql.types.DateType
+import org.jsoup.Jsoup
+
+
+import spark.implicits._ //!!!
+
+
+// COMMAND ----------
+
+case class Question(id: Long,
+                    ownerUserId: Long,
+                    creationDate: Timestamp,
+                    score: Long,
+                    title: String,
+                    body: String)
+
+val questionsDs = spark.read
+      .option("multiLine", true)
+      .option("header", true)
+      .schema(Encoders.product[Question].schema)
+      .option("escape", "\"")
+      .csv("/FileStore/tables/Questions.csv")
+      .na.drop("all")
+      .as[Question]
+      
+questionsDs.cache()
+
+
+// COMMAND ----------
+
+case class Answer(id: Long,
+                  ownerUserId: Long,
+                  creationDate: Timestamp,
+                  partnerId: Long,
+                  score: Long,
+                  body: String)
+
+val answersDs = spark.read
+      .option("multiLine", true)
+      .option("header", true)
+      .schema(Encoders.product[Answer].schema)
+      .option("escape", "\"")
+      .csv("/FileStore/tables/Answers.csv")
+      .na.drop("all")
+      .as[Answer]
+
+answersDs.cache()
+
+// COMMAND ----------
+
+case class Tag(id: Long,
+               tag: String)
+
+val tagsDs = spark.read
+      .option("multiLine", true)
+      .option("header", true)
+      .schema(Encoders.product[Tag].schema)
+      .option("escape", "\"")
+      .csv("/FileStore/tables/Tags.csv")
+      .na.drop("all")
+      .as[Tag]
+
+tagsDs.cache()
+
+// COMMAND ----------
+
+// MAGIC %md
+// MAGIC Collect daily statistics (number of unique users posted, number of questions and answers) into a separate table and export to a Parquet file.
+
+// COMMAND ----------
+
+val questionNaFreeDs = questionsDs
+      .withColumn("creationDate", col("creationDate").cast(DateType))
+      .na.drop(Seq("creationDate"))
+val answersNaFreeDs = answersDs
+  .withColumn("creationDate", col("creationDate").cast(DateType))
+  .na.drop(Seq("creationDate"))
+
+val questionCols = Seq(
+  lit(null).alias("answerId"),
+  col("ownerUserId"),
+  col("creationDate").cast(DateType),
+  col("id").alias("questionId")
+)
+val answerCols = Seq(
+  col("id").alias("answerId"),
+  col("ownerUserId"),
+  col("creationDate").cast(DateType),
+  lit(null).alias("questionId")
+)
+val unionDf = questionNaFreeDs.select(questionCols: _*).unionAll(answersNaFreeDs.select(answerCols: _ *))
```

Comment:
Experiment with unionByName as well.
We are using it now instead of simple union, b/c of some column ordering specifics